src/components/scrollManager.js

```diff
@@ -0,0 +1,501 @@
+
+    /**
+     * Ease value.
+     *
+     * @param {number} t value in range [0, 1]
+     * @return {number} eased value in range [0, 1]
+     */
+    function ease(t) {
+        return t*(2 - t); // easeOutQuad === ease-out
+    }
+
+    /**
+     * Document scroll wrapper helps to unify scrolling and fix issues of some browsers.
+     *
+     * webOS 2 Browser: scrolls documentElement (and window), but body has a scroll size
+     *
+     * webOS 3 Browser: scrolls body (and window)
+     *
+     * webOS 4 Native: scrolls body (and window); has a document.scrollingElement
+     *
+     * Tizen 4 Browser/Native: scrolls body (and window); has a document.scrollingElement
+     *
+     * Tizen 5 Browser/Native: scrolls documentElement (and window); has a document.scrollingElement
+     */
+    function DocumentScroller() {
+    }
+
+    DocumentScroller.prototype = {
+        get scrollLeft() {
+            return window.pageXOffset;
+        },
+        set scrollLeft(val) {
+            window.scroll(val, window.pageYOffset);
+        },
+
+        get scrollTop() {
+            return window.pageYOffset;
+        },
+        set scrollTop(val) {
+            window.scroll(window.pageXOffset, val);
+        },
+
+        get scrollWidth() {
+            return Math.max(document.documentElement.scrollWidth, document.body.scrollWidth);
+        },
+
+        get scrollHeight() {
+            return Math.max(document.documentElement.scrollHeight, document.body.scrollHeight);
+        },
+
+        get clientWidth() {
+            return Math.min(document.documentElement.clientWidth, document.body.clientWidth);
+        },
+
+        get clientHeight() {
+            return Math.min(document.documentElement.clientHeight, document.body.clientHeight);
+        },
+
+        getBoundingClientRect: function() {
+            // Make valid viewport coordinates: documentElement.getBoundingClientRect returns rect of entire document relative to viewport
+            return {
+                left: 0,
+                top: 0,
+                width: this.clientWidth,
+                height: this.clientHeight
+            };
+        },
+
+        scrollTo: function() {
+            window.scrollTo.apply(window, arguments);
+        }
+    };
+
+    var documentScroller = new DocumentScroller();
+
+    /**
+     * Returns parent element that can be scrolled. If no such, returns documentElement.
+     *
+     * @param {HTMLElement} element element for which parent is being searched
+     * @param {boolean} vertical search for vertical scrollable parent
+     */
+    function getScrollableParent(element, vertical) {
+        if (element) {
+            var parent = element.parentElement;
+
+            while (parent) {
+                if ((!vertical && parent.scrollWidth > parent.clientWidth && parent.classList.contains("scrollX")) ||
+                    (vertical && parent.scrollHeight > parent.clientHeight) && parent.classList.contains("scrollY")) {
```

Comment:
```suggestion
                    (vertical && parent.scrollHeight > parent.clientHeight && parent.classList.contains("scrollY"))) {
```
otherwise condition looks broken